### PR TITLE
test: switch description list tests to react testing library

### PR DESCRIPTION
**Pull request by lsprr**
Issue #1159

This PR switches the Line tests to React Testing Library.

### Validation steps

- Run yarn test
- See that there are coverage percentages in the table
- Verify that the tests are converted properly

### CODE REVIEW

File: packages/react/src/components/DescriptionList/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import {
+  DescriptionList,
+  DescriptionListItem,
+  DescriptionTerm,
+  DescriptionDetails
+} from './index';
```
Comment (scurker) [inline packages/react/src/components/DescriptionList/index.test.tsx:8]: Very minor nit.

```suggestion
} from './';
```

File: packages/react/src/components/DescriptionList/index.test.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+import axe from '../../axe';
+
+describe('DescriptionList components', () => {
+  test('renders a dl element', () => {
+    const { container } = render(<DescriptionList>a</DescriptionList>);
+    const dl = container.querySelector('dl');
+    expect(dl).toBeInTheDocument();
```
Comment (scurker) [inline packages/react/src/components/DescriptionList/index.test.tsx:15]: RTL typically recommends using a test id for instances like this:

```jsx
render(<DescriptionList data-testid="dl">a</DescriptionList>)
expect(screen.getByTestTestId('dl').tagName).toBe('DL')
```

File: packages/react/src/components/DescriptionList/index.test.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  });
+
+  test('renders a dt element', () => {
+    const { container } = render(<DescriptionTerm>a</DescriptionTerm>);
+    const dt = container.querySelector('dt');
+    expect(dt).toBeInTheDocument();
+  });
+
+  test('renders a dd element', () => {
+    const { container } = render(<DescriptionDetails>a</DescriptionDetails>);
```
Comment (scurker) [inline packages/react/src/components/DescriptionList/index.test.tsx:25]: Same as above, but I believe it should be `definition` as the role.

File: packages/react/src/components/DescriptionList/index.test.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    const dd = container.querySelector('dd');
+    expect(dd).toBeInTheDocument();
+  });
+
+  test('handles uncollapsed prop', () => {
+    const { container } = render(<DescriptionList>a</DescriptionList>);
+    const dl = container.querySelector('dl');
```
Comment (scurker) [inline packages/react/src/components/DescriptionList/index.test.tsx:32]: See above about `data-testid`

File: packages/react/src/components/DescriptionList/index.test.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    expect(dl).not.toHaveClass('DescriptionList--collapsed');
+  });
+
+  test('handles collapsed prop', () => {
+    const { container } = render(
```
Comment (scurker) [inline packages/react/src/components/DescriptionList/index.test.tsx:37]: See above about `data-testid`

File: packages/react/src/components/DescriptionList/index.test.tsx [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+      <DescriptionList collapsed>a</DescriptionList>
+    );
+    const dl = container.querySelector('dl');
+    expect(dl).toHaveClass('DescriptionList--collapsed');
+  });
+
+  test('passes classNames through', () => {
+    render(<DescriptionList className="a">a</DescriptionList>);
+    const dl = screen.getByText('a');
+    expect(dl).toHaveClass('a');
```
Comment (scurker) [inline packages/react/src/components/DescriptionList/index.test.tsx:47]: We should be sure that we're getting the right component here (by its role or test id) and that it doesn't clobber the existing class name that's already on the component.

**Review by scurker — COMMENTED** [review]
Just a few minor things to fix, thanks for your help here!

File: packages/react/src/components/DescriptionList/index.test.tsx (hunk 2 above)
Comment (lsprr) [reply packages/react/src/components/DescriptionList/index.test.tsx]: Thank you for the snippet! I wasn't aware of the tagname property until now. It's helpful for identifying HTML tags that don't have an assigned ARIA role.

**Comment by lsprr** [comment]
@scurker 

Thank you for the feedback and assistance. I greatly appreciate it!

**Review by scurker — APPROVED** [review]

**Comment by scurker** [comment]
🚨 Reviewed for Security 🚨

**Pull request merged**